pagarme/react-scripts-former-kit-dashboard
# Fix production build

This PR fixes a problem in the `webpack` configuration which creates a memory overflow in the node process to fix the problem was added a new dependency `babel-preset-react-app`.
With this fix, few minor updates were made in the dependencies.

## Context
- Fix production build

## How to test

1. Clone this project
```
git clone -b update/dependencies git@github.com:pagarme/react-scripts-former-kit-dashboard.git
```
2. Go to the project folder
```
  cd react-scripts-former-kit-dashboard/
```
3. Build a pack using the npm pack command
```
npm pack
```
4. In [Pilot project] change the `package.json` refence to `react-scripts-former-kit-dashboard` to the path for the created pack. 
![screen shot 2019-01-16 at 20 00 45](https://user-images.githubusercontent.com/1773766/51281404-71360f80-19c9-11e9-9ea7-5ab20ea21400.png)

5. In [Pilot project]  folder remove the `node_modules` and reinstall everything
```
  rm -rf node_modules/
  yarn
```
6. Run `yarn build`, in the [Pilot project](https://github.com/pagarme/pilot) using the branch [`update/react-scripts`](https://github.com/pagarme/pilot/tree/update/react-scripts)( which have all lint problems fixed )  and the project must work correctly.

7. Using [`serve`](https://www.npmjs.com/package/serve) you should be able to run Pilot with the command `serve`  inside the `build` folder created in the step `6`.

## Checklist
- [x] Build a production version of Pilot project without brake Storybook and the development buld
